alacritty_terminal/src/locale.rs

```diff
@@ -65,24 +65,20 @@ pub fn set_locale_environment() {
         result
     };
 
-    // Provide a default value for LC_CTYPE if needed
+    // provide a default value for LC_CTYPE if needed
     unsafe {
-        let orginal = setlocale(LC_CTYPE, null());
-        let ctype = if orginal.is_null() {
-            CString::new("C").unwrap()
+        // check if we already have LC_CTYPE set as a environment variable
+        if let Ok(ctype) = env::var("LC_CTYPE") {
+            setlocale(LC_CTYPE, CString::new(ctype).unwrap().as_ptr());
```

Comment:
```suggestion
            setlocale(LC_CTYPE, CString::new("").unwrap().as_ptr());
```

Reply:
> For example we could just always call setlocale with an empty string as parameter and check if the return value is C or not.

Why do you want to do this? I believe what we have right now is quite nice where we have one case to check if we have a environment variable, if we do we set the locale to it:

```rust
setlocale(LC_CTYPE, CString::new(ctype).unwrap().as_ptr());
```
If we don't have a enviorment variable passed along we will check if our `LC_CTYPE` locale is already set, and if it's `C` we will then overwrite it to `UTF-8`

```rust
let ctype = setlocale(LC_CTYPE, null());
if !ctype.is_null() && CStr::from_ptr(ctype).to_string_lossy() == "C" {
    let locale = "UTF-8";
    setlocale(LC_CTYPE, CString::new(locale).unwrap().as_ptr());
    env::set_var("LC_CTYPE", locale);
}
```

I am not sure i understand why you make [this change]( https://github.com/alacritty/alacritty/pull/3608/files#r414091716), could you elaborate on it @chrisduerr? I don't think we are missing any cases at the moment.

Reply:
Passing an empty string to `setlocale` tells it to set everything according to command line variables.

Reply:
What is the benefit of doing that rather than setting it to the passed variable? Care to elaborate on that @chrisduerr?